modules/log/log.go

```diff
@@ -174,7 +177,7 @@ type logMsg struct {
 // it can contain several providers and log message into all providers.
 type Logger struct {
 	adapter string
-	lock    sync.Mutex
+	lock    sync.RWMutex
```

Comment:
My point was, that reading the code, a lock called lock, with no comments, doesn't make it clear to me that it's a lock for outputs. I would add a comment, or name it outputsLock or something similar. But #1421 is also an option.